mlir/lib/Dialect/Vector/IR/VectorOps.cpp

```diff
@@ -5601,6 +5622,22 @@ LogicalResult ConstantMaskOp::verify() {
   return success();
 }
 
+bool ConstantMaskOp::isFullMask() {
+  auto resultType = getVectorType();
```

Comment:
I think @aartbik had already created some utilities to check if a mask is all ones or zeros.

Reply:
Do you happen to know where those utilities are? I poked around and couldn't find anything.

Reply:
I thought Aart would know by heart but I found it :) https://github.com/llvm/llvm-project/blob/main/mlir/lib/Dialect/Vector/IR/VectorOps.cpp#L56-L120

Reply:
Oh nice, although I don't think this would be something we should run as a folder, given that it might iterate over the elements of an `arith.constant` op. What about extracting out the constant mask (and maybe create_mask) handling alone and using it for the folder? Or we could add a canonicalization pattern instead.

Reply:
A canonicalization would be nice because we could turn an all-false transfer_read into a `vector.splat`, and for transfer_write just drop the op altogether. (Just catching up to this work, maybe some of it already exists)

Reply:
Yes, I would expect this to be a canonicalization.